fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/sink/FlinkSink.java

```diff
@@ -42,13 +43,9 @@
 import static com.alibaba.fluss.flink.sink.FlinkStreamPartitioner.partition;
 import static com.alibaba.fluss.flink.utils.FlinkConversions.toFlussRowType;
 
-/**
- * Flink sink for Fluss.
- *
- * <p>TODO: WithPreWriteTopology need to be changed to supportsPreWriteTopology in Flink 1.20. Trace
- * by https://github.com/alibaba/fluss/issues/622.
- */
-class FlinkSink implements Sink<RowData>, WithPreWriteTopology<RowData> {
+/** Flink sink for Fluss. */
+class FlinkSink
+        implements Sink<RowData>, SupportsPreWriteTopology<RowData>, WithPreWriteTopology<RowData> {
```

Comment:
I don't want to be dependent on a deprecated API as this API is removed in Flink 2.0, and we are planning to support Flink 2.0 in this version. 

A solution can be just dependent on `SupportsPreWriteTopology` here, and make the new introduced `SupportsPreWriteTopology` to implements `WithPreWriteTopology` in `fluss-flink-1.18`.